rollout/trafficrouting/istio/istio.go

```diff
@@ -255,6 +255,23 @@ func (r *Reconciler) reconcileVirtualService(obj *unstructured.Unstructured, vsv
 		if err = ValidateHTTPRoutes(r.rollout, vsvcRouteNames, httpRoutes); err != nil {
 			return nil, false, err
 		}
+
+		host, err := r.getDestinationRuleHost()
+		if err != nil {
+			return nil, false, err
+		}
+
+		if host != "" {
```

Comment:
Is it possible to have an empty host? We should check if Istio allows it.

Reply:
looks like the host is required for Istio [DestinationRule](https://istio.io/latest/docs/reference/config/networking/destination-rule/#DestinationRule). We can check do this check on the DestinationRule if included in `.spec.strategy.canary.trafficRouting.istio.destinationRule`

Reply:
ah that check already exist :) https://github.com/argoproj/argo-rollouts/blob/423be7b9ee37fade38a652b4fc6a43da8c842b5d/rollout/trafficrouting/istio/istio.go#L795